### PR TITLE
[FIXED] Freeze `n.Applied` for cluster when shutting down

### PR DESCRIPTION
**Pull request by MauriceVanVeen**
During shutdown calls to `js.processConsumerAssignment` can short-circuit with:
```go
	if s == nil || noMeta || shuttingDown {
		return
	}
```

Ensure that we don't up `n.Applied` and snapshot after while we're shutting down, as we can't be sure we've actually applied those entries.

Signed-off-by: Maurice van Veen <github@mauricevanveen.com>

### CODE REVIEW

File: server/jetstream_cluster.go [hunk 1]
```diff
@@ -1409,7 +1409,14 @@ func (js *jetStream) monitorCluster() {
 					continue
 				}
 				if didSnap, didStreamRemoval, didConsumerRemoval, err := js.applyMetaEntries(ce.Entries, ru); err == nil {
-					_, nb := n.Applied(ce.Index)
+					var nb uint64
+					js.mu.RLock()
+					shuttingDown := js.shuttingDown
```
Comment (derekcollison) [inline server/jetstream_cluster.go:1414]: I don't like that we need this Rlock here.

There is a convenience function here for this already

```
// Helper for determining if we are shutting down.
func (js *jetStream) isShuttingDown() bool {
	js.mu.RLock()
	defer js.mu.RUnlock()
	return js.shuttingDown
}
```

Should we make an atomic?

Comment (derekcollison) [reply server/jetstream_cluster.go:1414]: What does applied return for nb when shutting down?

Comment (MauriceVanVeen) [reply server/jetstream_cluster.go]: Am now using `js.isShuttingDown()` instead.

> What does applied return for nb when shutting down?

It returns the amount of total bytes currently in our WAL. Doesn't take into account if we're shutting down or not.
The important bit is that we should not be changing `n.applied` while `js.processConsumerRemoval` (for example) functions are failing due to `shuttingDown` and then a higher `n.applied` being used in a snapshot during shutdown. Which results in the `js.processConsumerRemoval` then not being replayed of course.

**Review by derekcollison — APPROVED** [review]
LGTM

**Pull request merged**